src/aai/aai.js

```diff
@@ -21,7 +21,24 @@
       var targetUrl = '';
       var opts = jQuery.extend({}, this.defaults, options),
           defaultCallback = function(e) {
-            targetUrl = opts.target + '?redirectUrl=' + window.location.href;
+            targetUrl = opts.target + '?redirectUrl=';
+            var redirectUrl = '';
+
+            // Redirection could be initiated from the login page; in that case,
+            // we need to retrieve the redirect URL from the URL parameters
+            var urlParams = new URLSearchParams(window.location.href.split('?')[1]);
+            var redirectUrlFromLogin = urlParams.get('redirectUrl') || null;
+
+            if (redirectUrlFromLogin == null || redirectUrlFromLogin === '') {
```

Comment:
var redirectUrl = window.location.href;
if (XXX)
{
   append...
}
would be more readable?

Reply:
Refactored